### ISSUE TITLE
Issues with the structure of ordered lists and their internal spaces and line breaks.

### ISSUE THREAD

**Issue by wyylg** [rvben/rumdl#577]
### rumdl version

0.1.76

### Installation method

Pre-built binary

### Bug description

Several problems occur: 
- the structure of nested ordered lists is broken.
- the leading spaces at the beginning of lines after an ordered list are modified inconsistently, with some lines having line breaks added.

### Steps to reproduce

1. test file:
````markdown
# aabbcc

## aabbcc

1. aabbcc
  aabbcc aabbcc aabbcc aabbcc.aabbcc aabbcc

  ```
  aabbcc (aabbcc) {
      'aabbcc' {
          aabbcc
      },
  }
  ```

2. aabbcc

  aabbcc aabbcc aabbcc

3. aabbcc
  aabbcc

  1. aabbcc
  2. aabbcc

---
### abc


````

2. run:
```
> rumdl fmt ./t1.md
t1.md:8:1: [MD077] Content after blank line in list item needs 3 spaces of indentation to remain part of the list
(found 2) [fixed]
t1.md:18:1: [MD077] Content after blank line in list item needs 3 spaces of indentation to remain part of the list (found 2) [fixed]
t1.md:20:1: [MD032] List should be followed by blank line [fixed]
t1.md:23:1: [MD005] Expected indentation of 0 spaces, found 2 [fixed]
t1.md:23:3: [MD029] Ordered list item number 1 does not match list style 'ordered' (expected 3)
t1.md:24:1: [MD005] Expected indentation of 0 spaces, found 2 [fixed]
t1.md:24:3: [MD029] Ordered list item number 2 does not match list style 'ordered' (expected 4)
t1.md:26:4: [MD065] Missing blank line after horizontal rule [fixed]
t1.md:27:1: [MD022] Expected 1 blank line above heading [fixed]
t1.md:28:1: [MD012] Multiple consecutive blank lines at end of file [fixed]

Fixed: Fixed 8/10 issues in 1 file (6ms)
>
```

### Expected behavior

````
# aabbcc

## aabbcc

1. aabbcc
   aabbcc aabbcc aabbcc aabbcc.aabbcc aabbcc

   ```
   aabbcc (aabbcc) {
       'aabbcc' {
           aabbcc
       },
   }
   ```

2. aabbcc

   aabbcc aabbcc aabbcc

3. aabbcc
   aabbcc

   1. aabbcc
   2. aabbcc

---

### abc

````

### Actual behavior

````
# aabbcc

## aabbcc

1. aabbcc
  aabbcc aabbcc aabbcc aabbcc.aabbcc aabbcc

   ```
   aabbcc (aabbcc) {
       'aabbcc' {
           aabbcc
       },
   }
   ```

2. aabbcc

   aabbcc aabbcc aabbcc

3. aabbcc

   aabbcc

4. aabbcc
5. aabbcc

---

### abc

````

### Configuration

```toml
> rumdl config --no-defaults
[global]
disable = ["MD041", "MD026", "MD040", "MD038"]          [from user config]
cache_dir = "C:/Users/u/AppData/Local/Temp/rumdl/cache" [from user config]

>
```

### Operating system

Windows

**Comment by rvben** [comment]
Fixed in [v0.1.81](https://github.com/rvben/rumdl/releases/tag/v0.1.81). Thanks for the report, @wyylg!
